docs/hints.md

```diff
@@ -5,6 +5,94 @@ in go-ceph work together. This is not meant to cover every possible use
 case but are recorded here as a quick way to get familiar with these
 calls.
 
+
+## General
+
+### Finding an API
+
+The go-ceph project wraps existing APIs that are part of Ceph. There are two
+kinds of APIs that are wrapped. The first style of API is based on functions
+Ceph exports as client libraries in C. The three packages in go-ceph `cephfs`,
+`rados`, and `rbd` map to `libcephfs`, `librados`, and `librbd` respectively.
+
+The go-ceph packages that wrap Ceph C APIs follow a documentation convention
+that aims to make it easier to map between the APIs. In functions that are
+implemented using a certain C API function a line with the term `Implements:`
+will be followed by the C function's declaration - matching what can be found
+in the C library's header file. For example, if you knew you wanted to wrap the
+rbd function to get an image's metadata, `rbd_metadata_get`, you could search
+within the source code or https://pkg.go.dev/github.com/ceph/go-ceph/rbd for
+`rbd_metadata_get` which would lead you to the `GetMetadata` method of the
+`Image` type.
+
+The second style of API is based on functions implemented within Ceph services
+based on Ceph's "command" system. These functions are primarily accessed using
+the `ceph` command line command. Many of the functions within the ceph command
+are implemented by sending a structured JSON message to either the Ceph MON or
+MGR. Packages in go-ceph that wrap these sorts of APIs are found in
+`cephfs/admin`, `rbd/admin`, and `common/admin/manager` for example.
+
+The command/JSON based API packages follow a different, but similar
+documentation convention to the C based APIs. Functions that roughly map to a
+particular `ceph` CLI command will contain a line with the term `Similar To:`
+followed by the `ceph` command line command it is similar to. For example, if
```

Comment:
```suggestion
followed by the `ceph` command line it is similar to. For example, if
```

Reply:
I prefer to keep the current wording. It seems somewhat redundant because in casual conversation we we call both the interface and the commands we type into it "Command line".
In "command line command" the first part is a qualifier saying what kind of command it is, distingushing it from other kinds of commands like a verbal command. :-)

As I write the above I am thinking I could possibly be convinced to call it a "shell command". Thoughts?

Reply:
If not how about:
```suggestion
followed by the `ceph` command it is similar to. For example, if
```